src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = MarketItemType.SUPPLY.toString().toLowerCase();
+    private static final String BUY_LABEL = MarketItemType.BUY.toString().toLowerCase();
+    private static final String RESULT_LABEL = "result";
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        MarketData marketData = new MarketData();
+
+        readFromFile(fromFileName, (line) -> {
+            String[] lineData = line.split(",");
+            String type = lineData[0].toLowerCase().trim();
+            int value = Integer.parseInt(lineData[1].trim());
+            if (type.equals(SUPPLY_LABEL)) {
+                marketData.addSupplyTotal(value);
+            } else if (type.equals(BUY_LABEL)) {
+                marketData.addBuyTotal(value);
+            }
+        });
```

Lines 18-27: The line-processing lambda assumes every line has two tokens and that the second token is an integer. There is no validation for empty or malformed lines; this can cause ArrayIndexOutOfBoundsException or NumberFormatException. The checklist says "#### Don't ignore exceptions" and shows that you should rethrow RuntimeException with helpful context; here you should validate the line (skip empty lines or ensure `tokens.length == 2`), and on parse errors throw a RuntimeException that includes the filename and the offending line so failures are diagnosable .